### DISCUSSION TITLE
Hotend fan noise optimization

### DISCUSSION THREAD

**Discussion by xblax** [xblax/flashforge_ad5m_klipper_mod#71] [General]
I think the hotend fan settings could be optimized, to lower the noise level a bit. By default it is running at 100% speed and not turned off until the nozzle cools to below 50°C:

```
[heater_fan hotend_fan]
pin: eboard:PB4
max_power: 1.0
#kick_start_time: 0.5
heater: extruder
heater_temp: 50.0
```

I think it should probably be enough to have the fan running at 70% and turn it off at 120°. Would like to hear your thoughts. 

I don't have any hard evidence behind these values. The reasoning is that air flow at 70% feels "good enough" and filament should not melt at 120°.

**Comment by KaruroChori** [comment]
Turning it off at 120°C can be risky.
There are filaments designed for low temperature printing, and if this is not understood by the user it would clog everything.
Still, it is also true we are preventing extrusion below 170°C, so those materials would not be usable regardless out of the box.

It is also unclear if the heat creep through the nozzle and the lack of airflow could heat up and deteriorate surrounding materials or parts, like the resin of the circuit board or example.

As for the fan speed, it would be ideal to get some thermal image at very low printing speeds as a worst case scenario.
Still, I would tune it down because it is too loud and the airflow seemed plenty to me as well :D.

**Reply by xblax** [reply]
With the nozzle sitting at 250° for 5min and the fan at 70% (no printing, no other fans), I can still touch the heat break and it barely feels warm at all. Based on that "metric" even 50% fan speed should work.

**Comment by xblax** [comment]
Does anyone here have an option on that topic?
For me it seems to work great with the hotend fan at 50% and temperature for disabling the hotend fan at 100°.

**Reply by consp** [reply]
I used to own some special filament which did not like long cooldowns for some reason (no material expert but it became brittle), is there a way to make it user configurable (e.g. as with the fan slider?). I like it when it cools down faster and with the pro box it's less of an issue.

**Reply by xblax** [reply]
For automatic enabled fans (heater_fan, controller_fan) there is not slider, unfortunately. But afaik the hotend fan does not contribute much to cooling the nozzle, the part cooling fan would be better suitable for that. It's only purpose is to keep the heat-break cool and prevent the filament from melting.

In my testing it stayed "cool" to touch even with the hotend fan running at 50% and the nozzle idling at 250° - but I can try to do some semi-scientific measurements with my infrared thermometer. Obviously if you have 70°C chamber temp for printing ABS on the Pro, the heat break would always be above that.

We can try to optimize the fan speed and turn off temp separately. 

For turn-off temp the questions are:
- How much does the hotend fan contribute to fast nozzle cooldown?
- How much of the remaining heat from the nozzle creeps into the heatbreak, after the hotend fan is off?

For fan speed the question is:
- How fast must the hotend fan run to prevent overheating of the heat-break during print?

**Reply by consp** [reply]
> But afaik the hotend fan does not contribute much to cooling the nozzle, 

Makes sense, the settings for that was "all fans on max". 

> Obviously if you have 70°C chamber temp for printing ABS on the Pro

Since there is not chamber heater it's only the hotplate doing the heating and there is no temperature sensor as well. Mine reached 50c at max (have a separate temp/humid sensor in there) which is fine for the filament I now use (bit warmer is good but it's no real issue).

> How much of the remaining heat from the nozzle creeps into the heatbreak, after the hotend fan is off?

Does it really matter after the glass transition temperature has been reached? I guess they used that as parameter as it's usually 60c for PLA and chose 50 to be safe.

> How fast must the hotend fan run to prevent overheating of the heat-break during print?

That's a good one, ambient temp definitely matters so the pro might be a bit different from the non-pro or one with a chamber. As well as if you live in iceland or death valley.
